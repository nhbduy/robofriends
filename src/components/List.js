import React, { Suspense } from 'react';

const Card = React.lazy(() => import('./Card'));

const List = ({ data }) => {
  return (
    <div>
      {data.map(item => (
        <Suspense key={item.id} fallback={<div>Loading...</div>}>
          <Card
            id={item.id}
            name={item.name}
            username={item.username}
            email={item.email}
          />
        </Suspense>
      ))}
    </div>
  );
};

export default List;
