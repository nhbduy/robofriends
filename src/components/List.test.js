import React from 'react';

import { shallow } from 'enzyme';
import List from './List';

it('expect to render List component', () => {
  const mockRobots = [
    { id: 1, name: 'John', username: 'john', email: 'john@gmail.com' }
  ];

  expect(shallow(<List data={mockRobots} />).debug()).toMatchSnapshot();
});
