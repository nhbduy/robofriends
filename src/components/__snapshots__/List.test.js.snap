// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render List component 1`] = `
"<div>
  <Suspense fallback={{...}}>
    <lazy id={1} name=\\"John\\" username=\\"john\\" email=\\"john@gmail.com\\" />
  </Suspense>
</div>"
`;
